src/beat_number.rs

```diff
@@ -0,0 +1,13 @@
+pub struct BeatNumber {
+    pub quarter_note: u32,
+    pub sixteenth_note: u32,
+}
+
+impl BeatNumber {
+    pub fn new(quarter_note: u32, sixteenth_note: u32) -> Self {
+        Self {
+            quarter_note,
+            sixteenth_note,
+        }
+    }
+}
```

Comment:
I think this is fine, but you don't technically need a constructor function for this struct since the fields are all public.

```rust
BeatNumber { quarter_note: 1, sixteenth_note: 2 }
```

Reply:
Ah nice, the non-constructor syntax is actually more descriptive, removed the constructor